src/main/java/com/thealgorithms/backtracking/SudokuSolver.java

```diff
@@ -0,0 +1,175 @@
+package com.thealgorithms.backtracking;
+
+/**
+ * Java program for Sudoku Solver using Backtracking Algorithm.
+ * Solves a given Sudoku puzzle by filling the empty cells (denoted by '0') with
+ * valid digits (1-9).
+ */
+public final class SudokuSolver {
```

Comment:
Looks like a similar class already exists. What's the difference between this?

https://github.com/TheAlgorithms/Java/blob/master/src/main/java/com/thealgorithms/others/Sudoku.java

You can refactor the existing one and add tests there. Also it would prefer if tests are parameterized, there are a lot of such examples in project.

Reply:
I wasn't aware there was already a Sudoku class in the `others` folder. I had only checked the `backtracking` folder, didn't find Sudoku there, and ended up implementing it in that location.

I will close this PR and open another one for refactoring the existing Sudoku code.